ios/MdocDataTransfer.swift

```diff
@@ -0,0 +1,195 @@
+
+    var bleServerTransfer: MdocGattServer?
+    var resolver: RCTPromiseResolveBlock?
+    var rejector: RCTPromiseRejectBlock?
+
+    override func supportedEvents() -> [String]! {
+        return ["onResponseSent", "onRequestReceived"]
+    }
+
+    // NFC is not enabled on iOS
+    @objc
+    func enableNfc() {
+        return
+    }
+
+    @objc
+    func initialize() -> String {
+        guard bleServerTransfer == nil else {
+            return ""
+        }
+
+        do {
+            bleServerTransfer = try MdocGattServer(parameters: [
+                InitializeKeys.document_json_data.rawValue: [],
+                InitializeKeys.trusted_certificates.rawValue: [],
+                InitializeKeys.send_response_manually.rawValue: true,
+            ])
+            bleServerTransfer?.delegate = self
+        } catch {
+            return error.localizedDescription
+        }
+
+        return ""
+    }
+
+    @objc(startQrEngagement:_:)
+    func startQrEngagement(
+        resolve: @escaping RCTPromiseResolveBlock,
+        reject: @escaping RCTPromiseRejectBlock
+    ) {
+        resolver = resolve
+        rejector = reject
+
+        guard let bleServerTransfer = bleServerTransfer else {
+            self.reject(
+                MdocDataTransferError.BleGattServerNotInitialized
+                    .localizedDescription)
+            return
+        }
+
+        bleServerTransfer.performDeviceEngagement()
+    }
+
+    @objc(sendDeviceResponse:)
+    func sendDeviceResponse(deviceResponse: String) -> String {
+        guard let bleServerTransfer = bleServerTransfer,
+            var sessionEncryption = bleServerTransfer.sessionEncryption
+        else {
+            return MdocDataTransferError.BleGattServerNotInitialized
+                .localizedDescription
+        }
+
+        do {
+            let byteArray = deviceResponse.split(separator: ":").compactMap {
+                UInt8($0)
+            }
+            let cipherData = try sessionEncryption.encrypt(byteArray)
+            let sd = SessionData(cipher_data: cipherData, status: 20)
+            try bleServerTransfer.sendResponse(
+                Data(sd.encode(options: CBOROptions())))
+        } catch {
+            let sd = SessionData(cipher_data: nil, status: 11)
+            do {
+                try bleServerTransfer.sendResponse(
+                    Data(sd.encode(options: CBOROptions())))
+            } catch {
+                return error.localizedDescription
+            }
+            return error.localizedDescription
+        }
+
+        return ""
+    }
+
+    @objc
+    func shutdown() -> String {
+        guard let bleServerTransfer = bleServerTransfer else {
+            return MdocDataTransferError.BleGattServerNotInitialized
+                .localizedDescription
+
+        }
+
+        bleServerTransfer.stop()
+
+        self.bleServerTransfer = nil
+        rejector = nil
+        resolver = nil
+
+        return ""
```

Comment:
again not sure on the empty string?

Reply:
empty string = no error
string = error

Could not find another pattern that works. Open to improvements in a follow up PR.

Reply:
nil is no error?